packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -358,47 +360,296 @@ class AppKitView extends _DarwinView {
+///
+/// ```dart
+/// import 'dart:js_interop';
+/// import 'dart:ui_web' as ui_web;
+/// import 'package:flutter/widgets.dart';
+///
+/// void useHostElement(BuildContext context) {
+///   final int flutterViewId = View.of(context).viewId;
+///   final JSAny? hostElement = ui_web.views.getHostElement(flutterViewId);
+///   // Use `package:web` with `hostElement`...
+/// }
+/// ```
+///
+/// **Important:** If your flutter App uses `registerViewFactory` to register a
+/// factory function that creates your `HtmlElementView` widget, take into
+/// account that that the `viewId` parameter passed to your factory function
+/// **is not the same** as the `flutterViewId` obtained above:
+///
+/// * `flutterViewId` (from `View.of(context)`) represents the [FlutterView]
+///   where the web app is currently rendering.
+/// * `viewId` (passed to the `viewFactory` function) represents a unique ID
+///   for the `HtmlElementView` instance that is being injected into the app.
+///
+/// Read more about [FlutterView] on Flutter's API docs:
+///
+/// * [`View.of`](https://api.flutter.dev/flutter/widgets/View/of.html)
+/// * [`getHostElement`](https://main-api.flutter.dev/flutter/dart-ui_web/FlutterViewManagerProxy/getHostElement.html)
+///
+/// ## Pointer events
+///
+/// `HtmlElementView` will interfere with Flutter's normal gesture detection. In
+/// order for the `HtmlElementView` contents to be interactive, they're allowed
+/// to handle `pointer-events`. This may result in Flutter missing some events
+/// because they've been handled by the `HtmlElementView`, and never seen by
+/// Flutter.
+///
+/// [`package:pointer_interceptor`](https://pub.dev/packages/pointer_interceptor)
+/// may help in some cases where Flutter content needs to be overlaid on top of
+/// an `HtmlElementView`. Alternatively, the `pointer-events: none` property can
+/// be set `onElementCreated`; but that will prevent **ALL** interactions with
+/// the underlying HTML content.
+///
+/// If the `HtmlElementView` is an `<iframe>` element, Flutter will not be able
+/// to see any pointer events that land in the `<iframe>` (click/tap, drag,
+/// drop...). In those cases, the HTML View will seem like it's "swallowing"
+/// the events and not participating in Flutter's gesture detection.
+///
+/// `package:webview_flutter` for the web is implemented as an `HtmlElementView`
+/// backed by an `<iframe>` element.
+///
+/// ## Visibility
+///
+/// `HtmlElementView` objects have an `isVisible` property that can be passed when
+/// registering the factory function, or calling `fromTagName`. In this case,
+/// the "visibility" refers to whether the `HtmlElementView` will paint pixels
+/// or not.
+///
+/// Correctly defining this value helps the Flutter web rendering engine optimize
+/// the amount of "overlays" it'll need to render a particular scene; some browsers
+/// can only have a limited number of overlays on a single page at a given time.
```

Comment:
It's safe to say "all" browsers. But more concretely, we should probably just say that overlays have performance cost both in memory and execution speed, and therefore it is best to reduce the number of overlays. `isVisible` is one way to do it.

Reply:
I extended this a little bit to introduce what an "overlay" is.